### DISCUSSION TITLE
Need suggestions for restaurant menu and food order

### DISCUSSION THREAD

**Discussion by coneforapine** [prisma/prisma#5292] [Q&A] [answered]
Hi I'm currently trying to create a relation between Food and Order models. What I'm trying to have is having `Food[]` inside Order.

The question is, how can I achieve such relationship with prisma?

**Comment by ryands17** [comment, marked as answer]
Hey @coneforapine 👋 

As a `Food` item can be inside multiple `Order` items and also `Order` items can contain multiple `Food` items, you would need a [many to many](https://www.prisma.io/docs/concepts/components/prisma-schema/relations#many-to-many-relations) relation for this. So something like the following:

```prisma
model Order {
  id Int @id @default(autoincrement())
  foodItems Food[]
}

model Food {
  id Int @id @default(autoincrement())
  orders Order[]
}
```

If you would like to store the quantity of the food item ordered as well, then you would need an explicit many-to-many relation as follows:

```prisma
model Order {
  id        Int          @id @default(autoincrement())
  foodItems FoodOrders[]
}

model Food {
  id     Int          @id @default(autoincrement())
  orders FoodOrders[]
}

model FoodOrders {
  id       Int    @id @default(autoincrement())
  quantity Int?   @default(1)
  order    Order? @relation(fields: [orderId], references: [id])
  orderId  Int?
  food     Food?  @relation(fields: [foodId], references: [id])
  foodId   Int?
}
```

Let me know if you have any other queries :)

**Reply by coneforapine** [reply]
Thanks for the answer. My other question is if you're planning to bring cascades anytime soon. I'm actually trying to switch from typeorm due to prisma's way of handling data and only missing thing for me is just cascades. I'm sure I can handle it with writing sql queries but it'd definitely be nice if we can just put it in our schema.

**Reply by ryands17** [reply]
Cascades is on our [roadmap](https://www.notion.so/Prisma-Roadmap-50766227b779464ab98899accb98295f) and will be worked on soon after Migrate is stable :)

As a workaround, you can use this community made tool that supports cascade deletes in your schema: https://paljs.com/plugins/delete

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
